### ISSUE TITLE
Make it work like a server instead of like a client

### ISSUE THREAD

**Issue by catqwe** [DevelopmentalOctopus/ha-buttplug#4]
Right now, this plugin works like a client, you have the Intiface in your phone connected to the toy, and then the plugin connects to it. But it just works if you are in the same network, so if I leave my home with the phone and the toy it does no longuer work, you could make workarounds using VPN but then they mess up with Android Auto, etc. Would be much easier to make the plugin work as a server open to the internet and the phone would connect to the public URL of HA.

And I understand that the new 'Repeater Mode' from Intiface allowes to pass the toy to another server.

I'm just not sure how to make it work with this plugin.

Any idea?

**Comment by catqwe** [comment]
I also asked this question in the buttplug forum: https://discuss.buttplug.io/t/make-repeater-mode-work-both-ways/327

**Comment by Bloempottt** [comment]
Did you find a solution for this? Having the same issue.

**Comment by catqwe** [comment]
No, I haven't found a solution and nobody answered me in the forum neither. @Bvdmgg Maybe add a comment there in the forum, it will get more visibility than here, seems like this project has been abandoned. "Repeater Mode" should allow this kind of usage, we have to figure out how to modify this component to adapt it to this new mode I guess.

**Comment by catqwe** [comment]
Maybe we could open an issue in this project also: https://github.com/intiface/intiface-engine

**Comment by catqwe** [comment]
The "Repeater Mode" just allows to pass a connection from one engine to another (like a proxy), but not a device, so it works in the inverted direction of what we need.

@Bvdmgg seems like the solution for this will be to have a Intiface Engine server running on another server, connect this home assistant integration to it.

And when this issue (https://github.com/intiface/intiface-central/issues/159) gets fixed, pass the device from the Intiface Central running in the phone to the Intiface Engine server. At that point it should work.

Until then the only option is to connect through a VPN connection.

**Comment by catqwe** [comment]
But making it work like a server would avoid the use of a server in the middle, so this feature request is still valid.

**Comment by MexHigh** [comment]
I think this is not how Home Assistant Integrations are intended to be used. But it should be possible to include intiface-central as a Home Assistant Add-On instead of an Integration. Haven't found one yet

**Comment by catqwe** [comment]
You are right @MexHigh Home Assistant Integrations should not work as servers. Two years ago I started to work in a Home Assitant Add-On https://github.com/catqwe/hassio-addons but realized that Intiface Central can't connect to https://github.com/intiface/intiface-engine until this issue gets fixed: https://github.com/intiface/intiface-central/issues/159

I guess I can close this issue since the work has to be done there.

**Comment by elliotcubit** [comment]
This addon I put together seems to work, though I've only tested it with Bluetooth. I literally just stuck `intiface-engine` in a docker container

https://github.com/elliotcubit/ha-intiface-addons

**Comment by catqwe** [comment]
@elliotcubit The problem that I'm trying to fix is to not having to connect to it locally, but have my device going on a walk in a park or in the car and having my home assistant instance at home.

We still need https://github.com/intiface/intiface-central/issues/159 to get fixed to accomplish that.
